schema/elements.go

```diff
@@ -30,7 +30,11 @@ type Schema struct {
 	once sync.Once
 	m    map[string]TypeDef
 
-	lock sync.Mutex
+	// Once used to protect the initialization of `lock` field.
+	lockOnce sync.Once
+	// Lock which protects writes to resolvedTypes. Used as pointer so that
+	// schema may be used as a value type
+	lock *sync.Mutex
```

Comment:
Do we need all these locks here? Can we make the lock by value since we're not supposed to copy a schema?

Reply:
yeah there were no unsafe schema usages in the ptr semantics refactor

Reply:
can remove the once. just need the lock